### PR TITLE
Refactor to composite GitHub Action with setup-java

### PR DESCRIPTION
**Pull request by ewels**
Refactor the repository to be a composite GitHub Action that first runs the `setup-java` action and then the custom action.

* **action.yml**
  - Add a new input `java-version` with a default value of `21`.
  - Change the action to use a composite run with steps.
  - Add a step to run `actions/setup-java` with the `java-version` input.
  - Add a step to run the custom action with the existing inputs.

* **.github/workflows/example.yml**
  - Remove the `setup-java` action step.
  - Ensure the workflow uses the updated composite action.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/nf-core/setup-nextflow/pull/174?shareId=c481e043-614b-467b-94e3-55d8df61d474).

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -10,9 +10,22 @@ inputs:
     description: "Whether to install every Nextflow release via the '-all' distribution."
     required: false
     default: false
+  java-version:
+    description: "The Java version to use. Example: 11"
```
Comment (ewels) [inline action.yml]:
```suggestion
    description: "The Java version to use."
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+### Added
+
+- Composite GitHub Action now includes `setup-java` action with `java-version` and `java-distribution` inputs
+- New input `java-version` with default value `21`
+- New input `java-distribution` with default value `zulu`
+
+### Changed
+
+- Custom action now runs after `setup-java` action
+- Removed `setup-java` action from `.github/workflows/example.yml` as it is now part of the composite action
```
Comment (ewels) [inline CHANGELOG.md]:
```suggestion
- Composite GitHub Action now includes `setup-java` action with `java-version` (default `21`) and `java-distribution` (default `zulu`) inputs

### Changed

- Removed `setup-java` action from `.github/workflows/example.yml` as it is now part of the composite action
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -98,6 +106,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
```
Comment (ewels) [inline CHANGELOG.md]:
```suggestion
```

File: action.yml [hunk 4]
```diff
@@ -3,16 +3,34 @@ description: "Install Nextflow and add it to the PATH"
...
-  main: "dist/index.js"
+  using: "composite"
+  steps:
+    - name: Setup Java
+      uses: actions/setup-java@v2
+      with:
+        java-version: ${{ inputs.java-version }}
+        distribution: ${{ inputs.java-distribution }}
+    - name: Run custom action
+      uses: ./
```
Comment (ewels) [inline action.yml]:
```suggestion
    - name: Install Nextflow
      run: node lib/src/main.js
```

**Comment by maxulysse** [comment]
> Don't recursively call yourself

Love the commit messages

**Comment by ewels** [comment]
Closing in favour of https://github.com/nf-core/setup-nextflow/pull/175

**Pull request closed without merging**